cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/ckan/CKANBackendImpl.java

```diff
@@ -516,25 +517,46 @@ public void capRecords(String orgName, String pkgName, String resName, long maxR
         // Get the resource ID by querying the cache
         String resId = cache.getResId(orgName, pkgName, resName);
         
-        // Get certain information about the records within the resource
-        JSONObject result = (JSONObject) getRecords(resId, null).get("result");
-        long total = (Long) result.get("total");
-        JSONArray records = (JSONArray) result.get("records");
-        
         // Create the filters for a datastore deletion
         String filters = "";
         
-        if (total > maxRecords) {
-            for (int i = 0; i < (total - maxRecords); i++) {
+        // Get the record pages, some variables
+        int offset = 0;
+        int limit = 100;
```

Comment:
Maybe this should be declared as constant (in this file if it is specific to this task or in a global place for constants in Cygnus if it has a wider effect).